### PR TITLE
adding option for load balanced apps

### PR DESCRIPTION
**Pull request by calellowitz**
@snopoke 
How does this look for the hq setup of load balanced read in main db apps?
`LOAD_BALANCED_APPS` would be a dict with form
```
{ 
"users": 
    {
     "pgmain": 5, 
     "pgmainstandby": 5
    }
}
```
The acceptable delay value for each db can be added to this dict when that is implemented

### CODE REVIEW

**Review by snopoke — COMMENTED** [review]
Looks like a good start. I think the logic should go into `ConnectionManager` which can also do validation on the setting configuration: https://github.com/dimagi/commcare-hq/blob/d8fdc0e5b2f7a3200ea743da60be8f808ddd8a60/corehq/sql_db/connections.py#L62

**Comment by calellowitz** [comment]
@snopoke I looked at the connection manager first, since that is where the analogous logic for the UCR databases is, but the doc string for `ConncectionManager` says it is for `dealing with sqlalchemy engines and sessions`. My understanding was that maindb lookups were not handled by sqlalchemy. Is that incorrect, or is the docstring just misleading.

File: corehq/sql_db/routers.py [hunk 1]
```diff
@@ -98,4 +100,13 @@ def db_for_read_write(model, write=True):
             engine_id = connection_manager.get_load_balanced_read_engine_id(ICDS_UCR_ENGINE_ID)
         return connection_manager.get_django_db_alias(engine_id)
     else:
+        if not write:
+            alias = connection_manager.get_load_balanced_read_engine_id(app_label)
+            cursor = connections[alias].cursor()
+            cursor.execute('SELECT now() - pg_last_xact_replay_timestamp() AS replication_delay;')
```
Comment (snopoke) [inline corehq/sql_db/routers.py]: @calellowitz FYI I've asked @sravfeyn to look into making a utility for doing this in a more genarlized way: https://trello.com/c/YD9iWTQU/377-utilities-to-monitor-replication-delay

**Comment by calellowitz** [comment]
@snopoke I started the commcare-cloud pr with the settings changes for this, but is this the hq change you had in mind. I was thinking of starting with accounting and user dbs since they get hit on almost every request and dont necessarily need to be super up to date, but was going to run through the other apps to see more candidates as well

**Comment by snopoke** [comment]
Yes, this is great though I don't think we want to check the replication delay on every query. I think we can probably remove that for now and @sravfeyn can add it back later.

File: corehq/sql_db/routers.py [hunk 2]
```diff
@@ -98,4 +100,13 @@ def db_for_read_write(model, write=True):
             engine_id = connection_manager.get_load_balanced_read_engine_id(ICDS_UCR_ENGINE_ID)
         return connection_manager.get_django_db_alias(engine_id)
     else:
+        if not write:
+            alias = connection_manager.get_load_balanced_read_engine_id(app_label)
```
Comment (snopoke) [inline corehq/sql_db/routers.py]: should rename this since it's now returning a django db alias as well

**Review by snopoke — APPROVED** [review]
Few small comments, nothing blocking

**Pull request merged**